### PR TITLE
GH-2324 inlining blank nodes

### PR DESCRIPTION
**Pull request by hmottestad**
GitHub issue resolved: #2324 

Briefly describe the changes proposed in this PR:

---- 
PR Author Checklist (see the [contributor guidelines](https://github.com/eclipse/rdf4j/blob/master/.github/CONTRIBUTING.md) for more details):

 - [ ] my pull request is [self-contained](https://rdf4j.org/documentation/developer/merge-strategy/#self-contained-changes-pull-requests-and-commits)
 - [ ] I've added tests for the changes I made
 - [ ] I've applied [code formatting](https://github.com/eclipse/rdf4j/blob/master/.github/CONTRIBUTING.md#code-formatting) (you can use `mvn process-resources` to format from the command line)
 - [ ] every commit message starts with the issue number (GH-xxxx) followed by a meaningful description of the change
 - [ ] every commit has been [signed off](https://stackoverflow.com/questions/1962094/what-is-the-sign-off-feature-in-git-for)

Note: we merge all feature pull requests using [squash and merge](https://help.github.com/en/github/administering-a-repository/about-merge-methods-on-github#squashing-your-merge-commits). See [RDF4J git merge strategy](https://rdf4j.org/documentation/developer/merge-strategy/) for more details.

### CODE REVIEW

**Comment by hmottestad** [comment]
@jeenbroekstra this fixes this bug, but it's a terrible solution. The issue is that there is no way to inform the delegate that we don't want to inline the object of a statement while at the same time telling it that we want to inline the subject. 

I feel that the overall handling on inlining is too split up, we have this code here which handles deciding if we should inline, and then we have some code somewhere else that does the actual inlining, and we keep a lot of stuff in memory but decide to build complex counters instead of just using a model an querying it while we are writing to decide what should be inlined.

I don't propose we do anything about it now though. I've already spent more time on this than I wanted, and at least now it outputs a correct result, even if it's not an optimal result.

File: core/rio/turtle/src/main/java/org/eclipse/rdf4j/rio/turtle/ArrangedWriter.java [hunk 1]
```diff
@@ -293,58 +299,74 @@ private boolean isStillReferenced(SubjectInContext key) {
...
-					BNode bNode = (BNode) obj;
-					if (!bNodeOccurence.containsKey(bNode)) {
-						bNodeOccurence.put(bNode, 0);
+					bNodeOccurrences.compute(
+							(BNode) obj,
+							(key, i) -> i == null ? 1 : i + 1
+					);
+
+					if (bNodeOccurrences.get(obj) > 1) {
+						noneInlinedNodes.add(st.getSubject());
```
Comment (jeenbroekstra) [inline core/rio/turtle/src/main/java/org/eclipse/rdf4j/rio/turtle/ArrangedWriter.java:338]: Shouldn't you check if the subject is a blank node before adding it?

Comment (jeenbroekstra) [reply core/rio/turtle/src/main/java/org/eclipse/rdf4j/rio/turtle/ArrangedWriter.java:338]: I've tweaked this locally and it seems to solve the inlining issue where you had to adjust the test case, without compromising on your own new test case. I'll push a change.

Comment (hmottestad) [reply core/rio/turtle/src/main/java/org/eclipse/rdf4j/rio/turtle/ArrangedWriter.java]: Not sure actually. But if it works better now than I'm fine with it.

**Pull request merged**